cantools/database/utils.py

```diff
@@ -102,52 +102,48 @@ def decode_data(data: bytes,
                 allow_truncated: bool,
                 ) -> SignalDictType:
 
-    unpacked = formats.big_endian.unpack(bytes(data), allow_truncated=allow_truncated)
-
-    if len(data) < expected_length and allow_truncated:
-        # to deal with truncated little-endian signals, we have to pad
-        # the raw data and remove the spurious signals after unpacking
-        # (i.e., before the signal values themselfs get decoded)
-        le_data = bytes(bytearray([0x00]*(expected_length - len(data)))
-                        + data[::-1])
-
-        le_unpacked = formats.little_endian.unpack(le_data)
-
-        # remove spurious little endian signals
+    actual_length = len(data)
+    if allow_truncated and actual_length < expected_length:
+        data = data.ljust(expected_length, b"\xFF")
+
+    unpacked = {
+        **formats.big_endian.unpack(bytes(data)),
```

Comment:
this causes the test failure you've stumbled over: big endian signals use sawtooth bit ordering, so it is really awkward to find out which bit ranges they occupy using a sane ordering. that was probably the reason why I separated the handling of BE and LE signals in the first place...

Reply:
(truncated BE can done directly by `bitstruct`)

Reply:
CANdb++ agrees with you. Interestingly it shows the startbit as 21 which to me makes it easier to reason about
![image](https://user-images.githubusercontent.com/52598363/171285925-40a11494-00b0-4b23-8640-e6411e838c10.png)

Reply:
@zariiii9003 Go to Options->Settings if this is not sufficiently confusing already :). I wonder why they left out the option to use roman numbering...

Reply:
@juleq that's true 😄 
But i fixed it and learned something new. Compared to master branch it is 4% slower with `decode(b'\x00\x11', decode_choices=False, scaling=False, allow_truncated=True)`. With scaling enabled the performance is equal to the master branch.
I can revert this though if you want to keep using the bitstruct parameter.

Reply:
> I can revert this though if you want to keep using the bitstruct parameter.

I vote for keeping your version because it uses the same mechanism for everything. (and it can use an older `bitstruct` version...)